articles/cloud-storage/cs-how-python-boto-ecs.md

```diff
@@ -0,0 +1,160 @@
+description: How to use Python and Boto with ECS
+services: cloud-storage
+author: Paul Brown
+toc_rootlink: How To
+toc_sub1: 
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: How to use Python and Boto with ECS
+toc_fullpath: How To/cs-how-python-boto-ecs.md
+toc_mdlink: cs-how-python-boto-ecs.md
+---
+
+# How to use Python and Boto with ECS
+
+## What is Boto?
+
+Boto is a Python package which enables interaction with UKCloud's Cloud Storage and will run on any client supporting Python.
+
+> [!NOTE]
+> This guide uses Python 3 and Boto 3 and has not been tested on previous versions.
+
+## Use cases
+
+You can use Boto to:
+
+- Automate Cloud Storage interactions, for example the creation and deletion of buckets, the uploading, downloading and deletion of objects.
+
+- Use Cloud Storage as a target for backups or long-term file retention.
+
+## Installing Python and Boto
+
+1. Navigate to the [Python homepage](https://www.python.org/) and install the version of Python 3 relevant for your OS.
+
+2. Next install the Boto 3 package via Python's Package Manager,  Pip.
+
+```Python
+pip install -U boto3
+```
+
+3. The following code will create a bucket, upload a file and display a percentage progress counter.
+
+```Python
+#!/usr/bin/env python3
+import os
+import sys
+import threading
+import boto3
+
+session = boto3.session.Session()
+class ProgressPercentage(object):
+    def __init__(self, filename):
+        """ Transfer progress percentage class """
+        self._filename = filename
+        self._size = float(os.path.getsize(filename))
+        self._seen_so_far = 0
+        self._lock = threading.Lock()
+    def __call__(self, bytes_amount):
+        with self._lock:
+            self._seen_so_far += bytes_amount
+            percentage = (self._seen_so_far / self._size) * 100
+            sys.stdout.write(
+                "\r%s  %s / %s  (%.2f%%)" % (
+                    self._filename, self._seen_so_far, self._size,
+                    percentage))
+            sys.stdout.flush()
+
+ukc_ecs_s3 = session.client(
+    service_name='s3',
+    # The following can be obtained from the UKCloud portal
+    aws_access_key_id='<access key or username>',
+    # The following can be obtained from the UKCloud portal
+    aws_secret_access_key='<secret key>',
+    # The endpoint will be either https://cas.frn00006.ukcloud.com
+    # or https://cas.frn00006.ukcloud.com
+    endpoint_url='<endpoint>',
+)
+
+# Assign source file, bucket name and key name values to vars
+source_file = '<source file directory and filename>'
+bucket_name = '<destination bucket>'
+# key name can be any value, suggest filename
+key_name = '<object key name>'
+
+# create bucket
+ukc_ecs_s3.create_bucket(Bucket=bucket_name)
+
+# Upload file
+ukc_ecs_s3.upload_file(source_file, bucket_name, key_name, Callback=ProgressPercentage(source_file))
+```
+
+4. The following code will download a file and display a percentage progress counter.
```

Comment:
Change to:
The following code downloads a file and displays a percentage progress counter.

Reply:
I think you missed this one ;)